### PR TITLE
Po event

### PR DESCRIPTION
**Pull request by petrie911**
Bg_Po_Event matched and documented.

### CODE REVIEW

**Comment by Dragorn421** [comment]
Nice, I noticed you commented out the data in .data.s instead of deleting the file
You should delete the file and remove it from the spec file
If the file is 100% OK you also want to use the compiled relocation data instead of the original
[See this other PR for how to edit the spec file](https://github.com/zeldaret/oot/commit/c9e85c4b8625386befb0bd801da69e35485c297c#diff-b979c2934ac0b4ba3f08dabfdd1b2299)

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:108]: can you use `i2 < 3` instead of `!= 3` ?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:117]: can you merge those conditions with an and `&&` ?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:126]: I think `12` makes more sense than `0xC` for a bitshift offset?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:127]: use NULL to check for null pointers such as the actor instance pointer returned by `Actor_SpawnAsChild`

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:144]: no need to put a `&` in front for function pointers

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:152]: Do you really have to use thisx over this?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:191]: can you avoid using thisx?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:254]: I think ` == 0` makes more sense when checking actual numerical values

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:287]: this is kinda weird, can you set  `phi_v1` and `phi_a1` outside of the sub-if/else,
or alternatively get rid of temp_f6 and temp_f10?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:311]: can this temp be inlined at all?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:314]: `Math_ApproxF` returns a bool so not having != 0 is ok

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:346]: use NULL to check for null pointers

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:405]: `Math_ApproxF` result -> bool

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:396]: constants on the right hand side? idk

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:442]: does `this->timer > 120` work?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:535]: I would remove many brackets but there's at least one which definitely has no purpose
`((func_8002DD78(player) != 0) && (thisx->xzDistFromLink < 320.0f))` -> `(func_8002DD78(player) != 0) && (thisx->xzDistFromLink < 320.0f)`

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:544]: can you merge this if into its parent? making an even bigger condition 😄 
It would allow to remove the return and wrap the rest of the function in an `else` clause though

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:553]: can you negate the condition and put the rest of the function under it?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:620]: line numbers make more sense in decimal

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:630]: we use decimal for colors

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.h
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.h:18]: why the 1 in collider1?

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
👍 nice
Do you have hindsights on how this actor works? the `thisx->child->child->child->child = thisx;` scares me although it looks like a great meme

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:254]: You'd think that, but it actually doesn't match. For modulo operations with a power of 4, setting it equal to zero and using logical not aren't the same thing. In the former, the compiler knows it doesn't have to care about the sign of the result, but it doesn't know that with logical not.

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:152]: No, but it's a lot shorter than having this->dyna.actor everywhere and the stack padding is needed.

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:191]: For some reason, at most one of those three params statements can use this->dyna.actor. If two or more do, it fails to match. I suppose I could leave the two thisx in the actortype and actorindex statements and switch the rest to this->dyna.actor

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:544]: Yes, and that's probably how it was written. super ugly.

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:311]: Not that I could figure. You can replace one, but if you replace both the compiler loads the pointer to firstFall first, then loads its value from the pointer. This makes very little sense.

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:126]: I'm not sure of this one. bitshifts are a very binary concept.

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:330]:
```suggestion
```

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:577]: since you put that here can it also be moved into Draw?

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
details

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:577]: No. In fact, since there's no inlines it doesn't even need to be moved down, so I could put it at the top with the rest of the data if that's preferred.

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:180]: This can easily be confused with the `ActorType` enum, maybe just `type`?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:469]:
```suggestion
        return false;
    }
    return true;
```

**Review by Dragorn421 — CHANGES_REQUESTED** [review]

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:559]: i can take care of this on my end if you want, but here is the prototype for this effect

`
void EffectSsDeadDb_Spawn(GlobalContext* globalCtx, Vec3f* pos, Vec3f* velocity, Vec3f* accel, s16 scale, s16 scaleStep, s16 primR, s16 primG, s16 primB, s16 primA, s16 envR, s16 envG, s16 envB, s16 unused, s32 arg14, s16 playSound)
`
sp54 is the position for the effect, and the 1 at the end is a boolean for if it should play a burning disintegrating sound. so colors/scale/scaleStep as decimal and that last thing as true would help

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:583]: s

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (fig02) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:585]: we refer to this is alpha everywhere else so we should keep that consistent

**Review by fig02 — COMMENTED** [review]
nice job, a few things

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:187]: it is.

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (petrie911) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c]: I seem to have run it on the whole repo inadvertently. this will add some changed files.

Comment (fig02) [reply src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c]: thats fine, I encourage it cause people forget to run it all the time

**Review by fig02 — APPROVED** [review]

**Comment by fig02** [comment]
still waiting on a roman review before a merge, but just a heads up that theres an undefined syms conflict

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:68]:
```suggestion
static u8 sBlockPosAtRest = 0;
```

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:319]: I guess "stop" stands for "stop falling" but I don't think it's the best name, the action is the main one where the player gets to push the blocks around (BlockPush), managing to solve it (BlockSolved) or not (BlockReset?)
so this is where I'd call this BlockMain but maybe BlockWaitInteract works too?

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:462]: just like BlockCheck isnt an action but has the Block prefix, should this be `PaintingNext`?

**Review by Dragorn421 — CHANGES_REQUESTED** [review]

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:189]:
```suggestion
    thisx->params &= 0x3F;
```

File: src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c:590]:
```suggestion
    s32 pad2;
```

**Review by Roman971 — APPROVED** [review]

**Pull request merged**
